themes-default/slim/views/addShows_addExistingShow.mako

```diff
@@ -24,89 +38,145 @@ const startVue = () => {
-                if (this.lastTxt === $('#rootDirText').val()) {
+                if (this.lastRootDirText === $('#rootDirText').val()) {
                     return false;
                 }
-                this.lastTxt = $('#rootDirText').val();
-                this.rootDirStaticList();
-                this.loadContent();
+                this.lastRootDirText = $('#rootDirText').val();
+                this.selectedRootDirs = this.rootDirs;
+                this.update();
             });
+
+            // Trigger Root Dirs to update.
+            // @FIXME: This is a workaround until root-dirs is a Vue component.
+            if ($('#rootDirText').val() === '') {
+                $('#rootDirs').click();
+            }
+
+            // this.lastRootDirText = $('#rootDirText').val();
+            // this.selectedRootDirs = this.rootDirs;
+            // this.update();
+        },
+        computed: {
+            rootDirs() {
+                if (!this.lastRootDirText) return [];
+                return this.lastRootDirText.split('|').slice(1);
+            },
+            filteredDirList() {
+                return this.dirList.filter(dir => !dir.alreadyAdded);
+            },
+            displayPaths() {
+                // Mark the root dir as bold in the path
+                return this.filteredDirList
+                    .map(dir => {
+                        const rootDir = this.rootDirs.find(rd => dir.path.startsWith(rd));
+                        const pathSep = rootDir.indexOf('\\\\') > -1 ? 2 : 1;
+                        const rdEndIndex = dir.path.indexOf(rootDir) + rootDir.length + pathSep;
+                        return '<b>' + dir.path.slice(0, rdEndIndex) + '</b>' + dir.path.slice(rdEndIndex);
+                    });
+            },
+            checkAll: {
+                get() {
+                    const selectedSeriesDirs = this.filteredDirList.filter(dir => dir.selected);
+                    if (!selectedSeriesDirs.length) return false;
+                    return selectedSeriesDirs.length === this.filteredDirList.length;
+                },
+                set(newValue) {
+                    this.dirList = this.dirList.map(dir => {
+                        dir.selected = newValue;
+                        return dir;
+                    });
+                }
+            }
         },
         methods: {
-            rootDirStaticList() {
-                $('#rootDirStaticList').html('');
-                $('#rootDirs option').each((index, element) => {
-                    $('#rootDirStaticList').append('<li class="ui-state-default ui-corner-all"><input type="checkbox" class="rootDirCheck" id="' + $(element).val() + '" checked="checked"> <label for="' + $(element).val() + '"><b>' + $(element).val() + '</b></label></li>');
+            async update() {
+                if (this.isLoading) return;
+
+                this.isLoading = true;
+                if (!this.selectedRootDirs.length) {
```

Comment:
If you're checking length you need to be providing a number to compare otherwise please remove the `.length`.